crates/rpc/rpc/src/eth/core.rs

```diff
@@ -450,9 +453,131 @@ where
     }
 }
 
+/// A helper to build the `EthApi` instance.
+#[derive(Debug)]
+pub struct EthApiBuilder<Provider, Pool, Network, EvmConfig>
+where
+    Provider: BlockReaderIdExt,
+{
+    provider: Provider,
+    pool: Pool,
+    network: Network,
+    evm_config: EvmConfig,
+    eth_cache: EthStateCache<Provider::Block, Provider::Receipt>,
+    gas_oracle: GasPriceOracle<Provider>,
+    gas_cap: GasCap,
+    max_simulate_blocks: u64,
+    eth_proof_window: u64,
+    blocking_task_pool: BlockingTaskPool,
```

Comment:
this I'd like to keep as Option just in case the user wants to provide a different pool,

we can do unwrap_or on build